### PR TITLE
Gamepadfix2

### PR DESCRIPTION
**Pull request by larsiusprime**

### CODE REVIEW

File: flixel/input/gamepad/FlxGamepad.hx [hunk 1]
```diff
@@ -629,7 +678,7 @@ enum FlxGamepadDeadZoneMode
 	 * The value of each axis is compared to the deadzone individually.
 	 * Works better when an analog stick is used like arrow keys for 4-directional-input.
 	 */
-	INDEPENDANT_AXES;
+	INDEPENDENT_AXES;
```
Comment (Gama11) [inline flixel/input/gamepad/FlxGamepad.hx]: Oops.

File: flixel/input/gamepad/id/FlxGamepadAnalog.hx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import flixel.input.gamepad.FlxGamepadInputID;
+import flixel.input.gamepad.FlxGamepad.FlxGamepadAnalogStick;
+
+/**
+ * A helper class for gamepad input.
+ * Provides optimized gamepad button checking using direct array access.
+ * 
+ */
+@:keep
+class FlxGamepadAnalog
```
Comment (Gama11) [inline flixel/input/gamepad/id/FlxGamepadAnalog.hx]: Maybe this should remain to be called "FlxGamepadAnalogList" and value should be "FlxGamepadAnalogValueList"?

File: flixel/input/gamepad/FlxGamepadManager.hx [hunk 3]
```diff
@@ -368,7 +368,7 @@ class FlxGamepadManager implements IFlxInputManager
 	@:allow(flixel.FlxG)
 	private function new() 
 	{
-		#if FLX_OPENFL_JOYSTICK_API
+		#if (FLX_OPENFL_JOYSTICK_API && !next)
```
Comment (Gama11) [inline flixel/input/gamepad/FlxGamepadManager.hx]: Just add that condition to `FlxConditionals` where `FLX_OPENFL_JOYSTICK_API` is defined.

Comment (Gama11) [inline flixel/input/gamepad/FlxGamepadManager.hx]: I think we might be able to replace all the `#if (!flash && !next)` conditions by that too, making things more readable?

File: flixel/input/gamepad/id/FlxGamepadAnalog.hx (hunk 2 above)
Comment (larsiusprime) [inline flixel/input/gamepad/id/FlxGamepadAnalog.hx]: Sure.

File: flixel/input/gamepad/FlxGamepadManager.hx (hunk 3 above)
Comment (larsiusprime) [inline flixel/input/gamepad/FlxGamepadManager.hx]: Yeah!

File: flixel/input/gamepad/id/FlxGamepadButtonList.hx [hunk 4]
```diff
@@ -10,23 +10,28 @@ import flixel.input.gamepad.id.FlxBaseGamepadList;
...
+	public var START            (get, never):Bool; inline function get_START()             { return check(FlxGamepadInputID.START);              }
+	public var LEFT_STICK_CLICK (get, never):Bool; inline function get_LEFT_STICK_CLICK()  { return check(FlxGamepadInputID.LEFT_STICK_CLICK);   }
+	public var RIGHT_STICK_CLICK(get, never):Bool; inline function get_RIGHT_STICK_CLICK() { return check(FlxGamepadInputID.RIGHT_STICK_CLICK);  }
+	public var GUIDE            (get, never):Bool; inline function get_GUIDE()             { return check(FlxGamepadInputID.GUIDE);              }
+	public var DPAD_UP          (get, never):Bool; inline function get_DPAD_UP()           { return check(FlxGamepadInputID.DPAD_UP);            }
+	public var DPAD_DOWN        (get, never):Bool; inline function get_DPAD_DOWN()         { return check(FlxGamepadInputID.DPAD_DOWN);          }
+	public var DPAD_LEFT        (get, never):Bool; inline function get_DPAD_LEFT()         { return check(FlxGamepadInputID.DPAD_LEFT);          }
+	public var DPAD_RIGHT       (get, never):Bool; inline function get_DPAD_RIGHT()        { return check(FlxGamepadInputID.DPAD_RIGHT);         }
+	#if(flash || next)
```
Comment (Gama11) [inline flixel/input/gamepad/id/FlxGamepadButtonList.hx]: Shouldn't the `#if` rather be in `FlxGamepadInputID`?

File: flixel/input/gamepad/FlxGamepadManager.hx [hunk 5]
```diff
@@ -469,7 +469,7 @@ class FlxGamepadManager implements IFlxInputManager
 	}
 	#end
 	
-	#if FLX_OPENFL_JOYSTICK_API
+	#if (FLX_OPENFL_JOYSTICK_API && !next)
```
Comment (Gama11) [inline flixel/input/gamepad/FlxGamepadManager.hx]: Leftover `!next`?

File: flixel/input/gamepad/id/XInputID.hx [hunk 6]
```diff
@@ -33,4 +33,16 @@ class XInputID
 	
 	public static inline var LEFT_TRIGGER:Int = 4;
 	public static inline var RIGHT_TRIGGER:Int = 5;
+	
+	#if (!next)
```
Comment (Gama11) [inline flixel/input/gamepad/id/XInputID.hx]: Why only `!next` here and not `!flash` too like everywhere else?

File: flixel/input/gamepad/id/FlxGamepadButtonList.hx (hunk 4 above)
Comment (larsiusprime) [inline flixel/input/gamepad/id/FlxGamepadButtonList.hx]: No, because then it's impossible to distinguish between LEFT_TRIGGER and LEFT_TRIGGER_FAKE.

**Pull request merged**